src/queues/processors/proposalFactory.ts

```diff
@@ -0,0 +1,62 @@
+import chunk from 'lodash.chunk';
+import { mailerQueue } from '../index';
+import { getFollows, getProposal } from '../../helpers/snapshot';
+import { getVerifiedSubscriptions } from '../../helpers/utils';
+import type { Job } from 'bull';
+
+function eventToTemplate(event: string) {
+  switch (event) {
+    case 'created':
+      return 'newProposal';
+    case 'end':
+      return 'closedProposal';
+    default:
+      throw new Error('Invalid proposal activity event type');
+  }
+}
+
+/**
+ * Return a list of email, for all subscribers following the given spaceId
+ */
+async function getSubscribersEmailFollowingSpace(spaceId: string) {
+  const subscriberEntries = await getVerifiedSubscriptions();
+  const subscriberKeyValuePairs: Iterable<[string, string]> = subscriberEntries.map(row => [
+    row.address as string,
+    row.email as string
+  ]);
+  const subscribers = new Map(subscriberKeyValuePairs);
+  const results = [];
+
+  // Batch the queries, as getFollows is limited to 1000 address per request
+  const addressesChunks = chunk(Array.from(subscribers.keys()), 1000);
+  for (const addressChunk of addressesChunks) {
+    const follows = await getFollows(addressChunk, spaceId);
+    results.push(...follows.map(follow => subscribers.get(follow.follower) as string));
+  }
+
+  return results;
+}
+
+export default async (job: Job): Promise<number> => {
+  const { event, id } = job.data;
+  const templateId = eventToTemplate(event);
+  const proposal = await getProposal(id);
```

Comment:
I think we can skip this part because we do it again [here](https://github.com/snapshot-labs/envelop/pull/53/files#diff-4c13ab9bfec45c969d95ab55903e1db931ac9f5701af3978d318350ed29e3dbeR7)

Reply:
If we skip that here, it will also skip the check on the next line, which check if proposal exist or not.

This security check ensure that when the proposal does not exist, we fail early here, instead of creating n jobs, and having this query run 1 time inside n jobs.